src/subdomains/supporting/payment/services/transaction-helper.ts

```diff
@@ -437,10 +437,21 @@ export class TransactionHelper implements OnModuleInit {
         return QuoteError.KYC_REQUIRED;
     }
 
+    const isSwapTx = isAsset(from) && isAsset(to);
+    if (isSwapTx) {
+      if (
+        (user?.userData.kycLevel < KycLevel.LEVEL_30 && user?.userData.status !== UserDataStatus.ACTIVE) ||
+        (txAmountChf > Config.tradingLimits.dailyDefault && user?.userData.kycLevel < KycLevel.LEVEL_50)
+      )
+        return QuoteError.KYC_REQUIRED;
+
+      if (txAmountChf > user?.userData.tradingLimit.remaining) return QuoteError.DEPOSIT_LIMIT_REACHED;
+    }
+
     if (paymentMethodIn === FiatPaymentMethod.INSTANT && user && !user.userData.olkypayAllowed)
       return QuoteError.KYC_REQUIRED_INSTANT;
     if (
-      ((isFiat(to) && to.name !== 'CHF') || paymentMethodIn === FiatPaymentMethod.CARD) &&
+      ((isFiat(to) && to.name !== 'CHF') || paymentMethodIn === FiatPaymentMethod.CARD || isSwapTx) &&
       user &&
       !user.userData.hasBankTxVerification &&
       txAmountChf > Config.tradingLimits.dailyDefault
```

Comment:
Move the amount checks up (above isSwapTx) and remove the `QuoteError.DEPOSIT_LIMIT_REACHED`. `(txAmountChf > Config.tradingLimits.dailyDefault && user?.userData.kycLevel < KycLevel.LEVEL_50)` can also be removed.